sw/device/lib/runtime/otbn.c

```diff
@@ -4,16 +4,109 @@
 
 #include "sw/device/lib/runtime/otbn.h"
 
+#include "sw/device/lib/base/freestanding/limits.h"
 #include "sw/device/lib/base/mmio.h"
 #include "sw/device/lib/dif/dif_base.h"
 #include "sw/device/lib/dif/dif_otbn.h"
+#include "sw/device/lib/irq.h"
+#include "sw/device/lib/runtime/ibex.h"
 #include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/rv_plic_testutils.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
 
 /**
  * Data width of big number subset, in bytes.
  */
 const int kOtbnWlenBytes = 256 / 8;
 
+/**
+ * Otbn context to be used by the isr handler, it should be assigned with the
+ * current otbn context pointer before enabling the irq.
+ */
+static otbn_t *otbn_irq_obj;
+
+/**
+ * Provides external IRQ handling for this test.
+ *
+ * This function overrides the default OTTF external ISR.
+ *
+ * It performs the following:
+ * 1. Claims the IRQ fired (finds PLIC IRQ index).
+ * 2. Checks that the index belongs to the OTBN peripheral.
+ * 3. Checks that the only IRQ is the `otbn_done`.
+ * 4. Clears the IRQ at the peripheral.
+ * 5. Completes the IRQ service at PLIC.
+ * 6. Set the `finished` flag to true.
+ */
+void ottf_external_isr(void) {
+  CHECK(otbn_irq_obj != NULL);
```

Comment:
What is the effect of throwing a failure inside the interrupt routine?

Reply:
I've changed the approach to make all the checkings outside the interrupt handler.